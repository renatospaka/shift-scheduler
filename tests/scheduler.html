
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>test: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/renatospaka/scheduler/adapter/event/test/mocEvent.go (100.0%)</option>
				
				<option value="file1">github.com/renatospaka/scheduler/adapter/event/test/mocListener.go (100.0%)</option>
				
				<option value="file2">github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go (80.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package test

// Mocked Event adapter
// NOT TO BE CALLED under no circumstances out of the testing process
type testEvent struct {
        data any
}

// Mocked method GetKey of an event
// NOT TO BE CALLED under no circumstances out of the testing process
func (e *testEvent) GetKey() string <span class="cov8" title="1">{
        return "event 1"
}</span>

// Mocked method GetData of an event
// NOT TO BE CALLED under no circumstances out of the testing process
func (e *testEvent) GetData() interface{} <span class="cov8" title="1">{
        e.data = "event 1"
        return e.data
}</span>
</pre>
		
		<pre class="file" id="file1" style="display: none">package test

type testListener struct {
        data   interface{}
        called bool
}

func (l *testListener) Handle() error <span class="cov8" title="1">{
        l.called = true
        return nil
}</span>

func (l *testListener) SetData(data interface{}) <span class="cov8" title="1">{
        l.data = data
}</span>
</pre>
		
		<pre class="file" id="file2" style="display: none">package tests

import (
        "context"
        "database/sql"
        "fmt"

        "github.com/renatospaka/scheduler/scheduler/worker/core/entity"
        doctoEntity "github.com/renatospaka/scheduler/scheduler/document/core/entity"
)

// Mocked Postgress repository
// NOT TO BE CALLED under no circumstances out of the testing process
type workerRepositoryMock struct {
        db *sql.DB
}

// Mocked method GetWorker of Worker's Repository
// NOT TO BE CALLED under no circumstances out of the testing process
func (w *workerRepositoryMock) GetWorker(id int) (*entity.Worker, error) <span class="cov8" title="1">{
        if id == 0 </span><span class="cov0" title="0">{
                return nil, nil
        }</span>

        <span class="cov8" title="1">worker, _ := entity.NewWorker(1, true, "Worker 1", "teacher")
        return worker, nil</span>
}


// Mocked method GetWorkerWithDocuments of Worker's Repository
// NOT TO BE CALLED under no circumstances out of the testing process
func (w *workerRepositoryMock) GetWorkerWithDocuments(id int) (*entity.Worker, error) <span class="cov8" title="1">{
        if id == 0 </span><span class="cov0" title="0">{
                return nil, nil
        }</span>

        <span class="cov8" title="1">worker, _ := entity.NewWorker(1, true, "Worker 1", "teacher")
        document1, _ := doctoEntity.NewDocument(71, true, "document 71")
        wrkerDocto1, _ := entity.NewWorkerDocument(1, document1)
        worker.LinkToDocument(wrkerDocto1)
        document2, _ := doctoEntity.NewDocument(2312, true, "document 2312")
        wrkerDocto2, _ := entity.NewWorkerDocument(2, document2)
        worker.LinkToDocument(wrkerDocto2)
        return worker, nil</span>
}

// Mocked method GetWorkerWithDocuments of Worker's Repository
// NOT TO BE CALLED under no circumstances out of the testing process
func (w *workerRepositoryMock) SetContext(ctx context.Context) <span class="cov0" title="0">{
        fmt.Println("Setting the Context")
}</span>
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
