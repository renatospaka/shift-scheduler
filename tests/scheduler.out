mode: set
github.com/renatospaka/scheduler/adapter/event/test/mocEvent.go:11.37,13.2 1 1
github.com/renatospaka/scheduler/adapter/event/test/mocEvent.go:17.43,20.2 2 1
github.com/renatospaka/scheduler/adapter/event/test/mocListener.go:8.39,11.2 2 1
github.com/renatospaka/scheduler/adapter/event/test/mocListener.go:13.50,15.2 1 1
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:20.74,21.13 1 1
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:21.13,23.3 1 0
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:25.2,26.20 2 1
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:32.87,33.13 1 1
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:33.13,35.3 1 0
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:37.2,44.20 8 1
github.com/renatospaka/scheduler/scheduler/worker/core/usecase/tests/mocRepository.go:49.64,51.2 1 0
